lib/book_settings.py

```diff
@@ -144,12 +143,7 @@ def search_for_asin(self, query):
         except Exception as e:
             try:
                 self._aConnection.close()
-                if self._proxy:
-                    self._aConnection = HTTPSConnection(self._https_address, self._https_port)
-                    self._aConnection.set_tunnel('www.amazon.com', 443)
-                else:
-                    self._aConnection = HTTPSConnection('www.amazon.com')
-
+                self._aConnection.connect()
```

Comment:
I was doing this because I ran into this exception called BadStatusLine (Raised if a server responds with a HTTP status code that we don’t understand according to https://docs.python.org/2/library/httplib.html). IIRC, I tried different ways of trying to get around it but closing and reopening seemed to be the only way to fix it. It only happened occasionally.